frontend/packages/console-dynamic-plugin-sdk/src/api/useResolvedExtensions.ts

```diff
@@ -43,42 +35,42 @@ import {
  */
 export const useResolvedExtensions = <E extends Extension>(
   ...typeGuards: ExtensionTypeGuard<E>[]
-): [ResolvedExtension<E>[], boolean, any] => {
+): [ResolvedExtension<E>[], boolean, any[]] => {
   const extensions = useExtensions<E>(...typeGuards);
 
   const [resolvedExtensions, setResolvedExtensions] = React.useState<ResolvedExtension<E>[]>([]);
   const [resolved, setResolved] = React.useState<boolean>(false);
-  const [error, setError] = React.useState<any>(undefined);
+  const [errors, setErrors] = React.useState<any[]>([]);
 
   React.useEffect(() => {
     let disposed = false;
 
-    Promise.all(
-      extensions.map(async (e) => {
-        const resolvedProperties = await resolveCodeRefProperties(e);
-        return mergeExtensionProperties(e, resolvedProperties) as ResolvedExtension<E>;
-      }),
-    )
-      .then((result) => {
-        if (!disposed) {
-          setResolvedExtensions(result);
-          setResolved(true);
-        }
-      })
-      .catch((err) => {
-        if (!disposed) {
-          setError(err);
+    // The promise returned by Promise.allSettled() never rejects; no need for catch-or-return.
+    // eslint-disable-next-line promise/catch-or-return
+    Promise.allSettled(
+      extensions.map((e) => resolveExtension<typeof e, any, ResolvedExtension<E>>(e)),
+    ).then((results) => {
+      if (!disposed) {
+        const [fulfilledValues, rejectedReasons] = unwrapPromiseSettledResults(results);
+        setResolvedExtensions(fulfilledValues);
+        setErrors(rejectedReasons);
+        setResolved(true);
+
+        if (rejectedReasons.length > 0) {
+          // eslint-disable-next-line no-console
+          console.error('Detected errors while resolving Console extensions', rejectedReasons);
         }
-      });
+      }
+    });
 
     return () => {
       disposed = true;
     };
   }, [extensions]);
 
-  return [resolvedExtensions, resolved, error];
+  return React.useMemo(() => [resolvedExtensions, resolved, errors], [
```

Comment:
you dont need to wrap it in useMemo. All values are from state, thus they dont change unless we call setters

Reply:
This was done to comply with current JSDoc comment which states

> The hook's result is guaranteed to be referentially stable across re-renders.

But you're right, people usually destructure the hook's result anyway, so no need for memo there.

Reply:
Updated the hook's JSDoc comment accordingly.